### PR TITLE
Optimize Copilot and CodeRabbit PR review configs

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Added global Copilot instructions file (`.github/copilot-instructions.md`) covering
  project context, focus areas, Rust conventions, known patterns, and review priorities.
- Added path-scoped Copilot instruction files for error handling (`rust-error-handling`)
  and test code review (`rust-tests`).
- Enhanced CodeRabbit with `path_instructions` covering server/HTTP client code, config
  loading, rating workflows, wizard input validation, and general Rust patterns --
  areas where analysis showed CodeRabbit underperformed.
- Enabled CodeRabbit tool integrations: shellcheck, actionlint, yamllint, gitleaks.
- Added CodeRabbit `knowledge_base` with `code_guidelines` to auto-learn review
  standards from CLAUDE.md and copilot-instructions.md.
- Copilot `review_on_push` disabled in the repository ruleset (separate from this PR)
  to eliminate the 61% repeat noise rate in later review rounds.

## Context

Analysis of 18 merged PRs in this repo showed:
- Copilot produces 2-3x more inline comments than CodeRabbit, but 20-30% are repeats
  on multi-push PRs
- Round 2 is ~50% signal, Round 3+ is ~65% noise
- CodeRabbit and Copilot find different things (low overlap), making them complementary
- CodeRabbit's gaps (error path edge cases, cross-platform issues, test coverage) are
  addressable via path_instructions configuration

## Test plan

- [ ] Verify Copilot reads global + path-scoped instruction files on next PR
- [ ] Verify CodeRabbit applies path_instructions on next PR
- [ ] Verify CodeRabbit tool integrations produce output

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Chores**
  * Expanded automated review configuration with path-scoped instruction blocks, added additional linting/security tools (shellcheck, actionlint, yamllint, gitleaks), and tightened checks for config, env, and CLI-related issues; preserved a required review-summary footer.
* **Documentation**
  * Added comprehensive PR-review guidance covering security, error-handling, test isolation, API/cross-platform correctness, and prioritized review criteria to improve review consistency and code quality.

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  transient errors, error body truncation inconsistencies
+- API correctness: URL encoding for user-provided IDs, header injection, response
+  body handling
+- Cross-platform issues: path separators, case sensitivity, hostname handling
+
+## Rust conventions
+
+- Prefer `thiserror` for error enum definitions
+- Prefer explicit error handling over `.unwrap()` in non-test code
+- Prefer `?` with context via `.map_err()` or `anyhow::Context`
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply .github/copilot-instructions.md]: Fixed in fe3cc36. Replaced thiserror/anyhow references with the actual pattern: hand-rolled error enums with manual Display/Error impls, and contextual errors via .map_err() with project-specific error types.

File: .github/instructions/rust-tests.instructions.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+applyTo: "smpr/src/**/*test*"
+excludeAgent: "coding-agent"
+---
+
+# Test Code Review
+
+- Tests run with `--test-threads=1` because config tests mutate process-global
+  env vars. Verify test isolation where env vars are set and cleared.
+- `#[ignore]` tests require a live media server (UAT). Verify they do not
+  silently pass when the server is unreachable.
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply .github/instructions/rust-tests.instructions.md]: Fixed in cdb8aa8. Updated to describe SMPR_UAT_TEST=1 env var with early return pattern, including verification that skip message is printed.
